OpenRA.Mods.Common/LoadScreens/BlankLoadScreen.cs

```diff
@@ -78,7 +80,14 @@ public void StartGame(Arguments args)
 			var replay = args != null ? args.GetValue("Launch.Replay", null) : null;
 			if (!string.IsNullOrEmpty(replay))
 			{
-				Game.JoinReplay(replay);
+				var replayMeta = ReplayMetadata.Read(replay);
+				if (ReplayUtils.CheckReplayCompatibility(replayMeta, Game.LoadShellMap))
+					Game.JoinReplay(replay);
+
+				var mod = replayMeta.GameInfo.Mod;
```

Comment:
This code still runs and throws after `CheckReplayCompatibility` displays its error dialog.